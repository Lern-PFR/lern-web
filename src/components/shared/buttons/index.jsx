import PropTypes from 'prop-types';
import {
	primary,
	standard,
	subtle,
	dropdown,
	danger,
	iconButton,
	floatingActionButton,
	primaryLink,
	outlined,
	subtleLink,
} from 'theme/buttonStyles';
import DynamicButtonComponent from './DynamicButtonComponent';
import DynamicLinkButtonComponent from './DynamicLinkButtonComponent';

// @TODO: write link and subtleLink buttons once navigation has been set up.

/**
 * @name PrimaryButton
 * @description Button component to use as Call to action. Should only be used once per page (header not included).
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const PrimaryButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...primary} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name PrimaryLinkButton
 * @description Link button component to use as Call to action. Should only be used once per page (header not included).
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const PrimaryLinkButton = ({ children, ...otherProps }) => (
	<DynamicLinkButtonComponent linkStyle={primaryLink} {...primary} {...otherProps}>
		{children}
	</DynamicLinkButtonComponent>
);

/**
 * @name OutlinedButton
 * @description Button component to use as primary action when a CTA is already defined.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const OutlinedButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent linkStyle={primaryLink} {...outlined} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name OutlinedLinkButton
 * @description Link button component to use as primary when a CTA is already defined.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children The children to display inside this button.
 */
const OutlinedLinkButton = ({ children, ...otherProps }) => (
	<DynamicLinkButtonComponent linkStyle={primaryLink} {...outlined} {...otherProps}>
		{children}
	</DynamicLinkButtonComponent>
);

/**
 * @name StandardButton
 * @description Standard button component to be used for most use-cases.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const StandardButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...standard} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name SubtleButton
 * @description Button component to be used along a primary button for less crucial actions such as "cancel".
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const SubtleButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...subtle} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name SubtleLinkButton
 * @description Link button component to be used along a primary button for less used redirections.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const SubtleLinkButton = ({ children, ...otherProps }) => (
	<DynamicLinkButtonComponent linkStyle={subtleLink} {...subtle} {...otherProps}>
		{children}
	</DynamicLinkButtonComponent>
);

/**
 * @name DropdownButton
 * @description Button component to be used for dropdown menu trigger.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const DropdownButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...dropdown} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name DangerButton
 * @description Button component to use for final confirmation to a destructive action such as deleting data.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const DangerButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...danger} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name IconButton
 * @description Button component intended to display only the icon it contains, but behave as a regular button.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const IconButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...iconButton} {...otherProps} isIconOnly>
		{children}
	</DynamicButtonComponent>
);

/**
 * @name FloatingActionButton
 * @description Button component intended to display a floating action button.
 *
 * @author Timothée Simon-Franza
 *
 * @param {*} children : The children to display inside this button.
 */
const FloatingActionButton = ({ children, ...otherProps }) => (
	<DynamicButtonComponent {...floatingActionButton} {...otherProps}>
		{children}
	</DynamicButtonComponent>
);

PrimaryButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

PrimaryLinkButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

OutlinedButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

OutlinedLinkButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

StandardButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

SubtleButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

SubtleLinkButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

DropdownButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

DangerButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

IconButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

FloatingActionButton.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

export {
	FloatingActionButton,
	OutlinedButton,
	OutlinedLinkButton,
	PrimaryButton,
	PrimaryLinkButton,
	StandardButton,
	SubtleButton,
	SubtleLinkButton,
	DropdownButton,
	DangerButton,
	IconButton,
};
