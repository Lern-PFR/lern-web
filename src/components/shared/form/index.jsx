import FieldsetComponent from './FieldsetComponent';
import LabeledCheckbox from './LabeledCheckbox';
import LabeledRadioButton from './LabeledRadioButton';
import LabelComponent from './LabelComponent';
import LabeledInput from './LabeledInput';

export {
	LabeledCheckbox,
	FieldsetComponent,
	LabeledRadioButton,
	LabelComponent,
	LabeledInput,
};
