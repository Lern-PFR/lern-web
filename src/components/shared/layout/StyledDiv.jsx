import styled from 'styled-components';
import {
	color,
	flexbox,
	layout,
	margin,
	padding,
	position,
	shadow,
} from 'styled-system';

/**
 * @name StyledDiv
 * @description A reusable div component that can be styled using styled-components and styled-system.
 *
 * @author TimothÃ©e Simon-Franza
 */
const StyledDiv = styled('div')(
	{
		visibility: (({ visibility }) => (visibility || 'initial')),
		boxSizing: (({ boxSizing }) => (boxSizing || 'border-box')),
	},
	color,
	flexbox,
	layout,
	margin,
	padding,
	position,
	shadow,
);

export default StyledDiv;
