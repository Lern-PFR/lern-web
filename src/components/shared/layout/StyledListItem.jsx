import styled from 'styled-components';
import {
	border,
	color,
	flexbox,
	grid,
	layout,
	margin,
	padding,
	position,
	shadow,
} from 'styled-system';

/**
 * @name StyledListItem
 * @description A reusable li component that can be styled using styled-components and styled-system.
 *
 * @author TimothÃ©e Simon-Franza
 */
const StyledListItem = styled('li')(
	{
		visibility: (({ visibility }) => (visibility || 'initial')),
		boxSizing: (({ boxSizing }) => (boxSizing || 'border-box')),
		zIndex: (({ zIndex }) => (zIndex || 'initial')),
	},
	border,
	color,
	flexbox,
	grid,
	layout,
	margin,
	padding,
	position,
	shadow,
);

export default StyledListItem;
