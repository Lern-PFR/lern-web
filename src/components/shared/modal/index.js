import Modal from './Modal';
import ModalFooter from './ModalFooter';
import PromptModal from './PromptModal';
import useModal from './useModal';

export {
	Modal,
	ModalFooter,
	PromptModal,
	useModal,
};
