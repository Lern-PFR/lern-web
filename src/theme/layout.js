import { mainLayoutTopPadding } from './constants';

const mainContainerLayout = {
	display: 'flex',
	flexDirection: 'column',
	overflow: 'auto',
	height: '100%',
	width: '100%',
};

const mainLayout = {
	flex: '1',
	padding: `${mainLayoutTopPadding}px`,
};

export {
	mainContainerLayout,
	mainLayout,
};
