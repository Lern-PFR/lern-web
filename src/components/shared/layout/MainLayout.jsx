import PropTypes from 'prop-types';
import { mainContainerLayout, mainLayout } from 'theme/layout';
import { Navbar } from 'components/shared/navigation';
import { StyledDiv } from 'components/shared/styledElements';

/**
 * @name MainLayout
 * @description The main layout to use for every page of the application.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {*} children Elements to be displayed in the page.
 */
const MainLayout = ({ children }) => (
	<StyledDiv {...mainContainerLayout}>
		<Navbar />
		<StyledDiv as="main" {...mainLayout}>
			{children}
		</StyledDiv>
	</StyledDiv>
);

MainLayout.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.elementType,
		PropTypes.node,
	]).isRequired,
};

export default MainLayout;
