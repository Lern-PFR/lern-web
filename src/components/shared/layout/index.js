import MainLayout from './MainLayout';

export {
	// eslint-disable-next-line import/prefer-default-export
	MainLayout,
};
