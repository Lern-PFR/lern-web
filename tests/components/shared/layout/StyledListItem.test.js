import { shallow } from 'enzyme';
import { StyledListItem } from 'components/shared/layout';

describe('StyledListItem', () => {
	it('should match previous snapshot', () => {
		const sut = (<StyledListItem>Hello, world!</StyledListItem>);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
