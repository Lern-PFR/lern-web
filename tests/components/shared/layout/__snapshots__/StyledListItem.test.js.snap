// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledListItem should match previous snapshot 1`] = `
<li
  className="sc-dlfnbm lNSRg"
>
  Hello, world!
</li>
`;
